src/domains/transaction/hooks/use-transaction-builder.ts

```diff
@@ -16,10 +18,27 @@ const prepareMultiSignature = async (
 		.multiSignature(wallet.multiSignature().all().min, wallet.multiSignature().all().publicKeys),
 });
 
-const prepareLedger = async (input: Services.TransactionInputs, wallet: ProfileContracts.IReadWriteWallet) => ({
-	...input,
-	signatory: await wallet.signatory().ledger(wallet.data().get<string>(ProfileContracts.WalletData.DerivationPath)!),
-});
+const prepareLedger = async (
+	profile: ProfileContracts.IProfile,
+	input: Services.TransactionInputs,
+	wallet: ProfileContracts.IReadWriteWallet,
+	connectFn: ConnectFn,
+	transport: typeof Transport,
+) => {
+	await connectFn(profile, wallet.coinId(), wallet.networkId());
+
+	await wallet.ledger().connect(transport);
```

Comment:
What's the reason for connecting here? The SDK will now automatically connect to the ledger and also disconnect from it after it has signed the transaction.

Reply:
Right, is aborting the connection handled as well?

Reply:
Not sure what you mean by aborting? We connect, sign the transaction and then disconnect.

Reply:
Ah I mean if this https://github.com/ArkEcosystem/desktop-wallet/pull/3846/files#diff-f7cc672ff8254a45a8770d12189d9ac7d7b7a38c53b2ccfc2f03a06d1fbf14e1R76 will play well because I haven't tested in dev mode

Reply:
This should probably now instead wrap the `signFn` call which will connect, sign and disconnect all in one go instead of having to call a lot of separate functions.

Reply:
@faustbrian it appears that the connection needs to run through `useLedgerConnection` https://github.com/ArkEcosystem/desktop-wallet/blob/develop/src/app/contexts/Ledger/hooks/connection.ts#L52 because it updates internal states that are required to render the ledger review step.  For the context of this PR, I only moved the connect methods higher than `transactionBuilder` for better readability, and we can focus to clean and simplify this particular code on a follow-up PR.